### DISCUSSION TITLE
v3.0.0-alpha.2

### DISCUSSION THREAD

**Discussion by adamwathan** [tailwindlabs/tailwindcss#6012] [Announcements]
To upgrade, install `tailwindcss@next`:

```
npm install -D tailwindcss@next
```

If you're using [@tailwindcss/typography](https://github.com/tailwindlabs/tailwindcss-typography), you'll want to update that package as well:

```
npm install -D @tailwindcss/typography@next
```

**Remember this is an alpha release**, so some things might be broken. I don't know about them otherwise I would have fixed them, but you are gonna find them! So try it, break it, and tell me about it when you do so we can get this thing really battle-tested for a proper v3.0 release in a few weeks.

---

### Changed

- Don't use pointer cursor on disabled buttons by default ([#5772](https://github.com/tailwindlabs/tailwindcss/pull/5772))
- Set default content value in preflight instead of within each before/after utility ([#5820](https://github.com/tailwindlabs/tailwindcss/pull/5820))
- Remove `prefix` as a function ([#5829](https://github.com/tailwindlabs/tailwindcss/pull/5829))

### Added

- Add `flex-basis` utilities ([#5671](https://github.com/tailwindlabs/tailwindcss/pull/5671))
- Make negative values a first-class feature ([#5709](https://github.com/tailwindlabs/tailwindcss/pull/5709))
- Add `fit-content` values for `min/max-width/height` utilities ([#5638](https://github.com/tailwindlabs/tailwindcss/pull/5638))
- Add `min/max-content` values for `min/max-height` utilities ([#5729](https://github.com/tailwindlabs/tailwindcss/pull/5729))
- Add all standard `cursor-*` values by default ([#5734](https://github.com/tailwindlabs/tailwindcss/pull/5734))
- Add `grow-*` and `shrink-*` utilities, deprecate `flex-grow-*` and `flex-shrink-*` ([#5733](https://github.com/tailwindlabs/tailwindcss/pull/5733))
- Add `text-decoration-color` utilities ([#5760](https://github.com/tailwindlabs/tailwindcss/pull/5760))
- Add new declarative `addVariant` API ([#5809](https://github.com/tailwindlabs/tailwindcss/pull/5809))
- Add first-class `print` variant for targeting printed media ([#5885](https://github.com/tailwindlabs/tailwindcss/pull/5885))
- Add full color palette for `fill-*` and `stroke-*` utilities (#5933[](https://github.com/tailwindlabs/tailwindcss/pull/5933))
- Add composable API for colored box shadows ([#5979](https://github.com/tailwindlabs/tailwindcss/pull/5979))

### Fixed

- Configure chokidar's `awaitWriteFinish` setting to avoid occasional stale builds on Windows ([#5774](https://github.com/tailwindlabs/tailwindcss/pull/5774))
- Fix CLI `--content` option ([#5775](https://github.com/tailwindlabs/tailwindcss/pull/5775))
- Fix before/after utilities overriding custom content values at larger breakpoints ([#5820](https://github.com/tailwindlabs/tailwindcss/pull/5820))
- Cleanup duplicate properties ([#5830](https://github.com/tailwindlabs/tailwindcss/pull/5830))
- Allow `_` inside `url()` when using arbitrary values ([#5853](https://github.com/tailwindlabs/tailwindcss/pull/5853))
- Prevent crashes when using comments in `@layer` AtRules ([#5854](https://github.com/tailwindlabs/tailwindcss/pull/5854))
- Handle color transformations properly with `theme(...)` for all relevant plugins ([#4533](https://github.com/tailwindlabs/tailwindcss/pull/4533), [#5871](https://github.com/tailwindlabs/tailwindcss/pull/5871))
- Ensure `@apply`-ing a utility with multiple definitions works ([#5870](https://github.com/tailwindlabs/tailwindcss/pull/5870))

---

Thanks to everyone who contributed to this release: @adamwathan, @bradlc, @dance2die, @dcastil, @DoctorDerek, @hardfist, @lukewarlow, @MatteoGauthier, @MichaelAllenWarner, @nifte, @reinink, @RobinMalfait, @sachinraja, @xzfd1010 ❤️

<hr /><em>This discussion was created from the release <a href='https://github.com/tailwindlabs/tailwindcss/releases/tag/v3.0.0-alpha.2'>v3.0.0-alpha.2</a>.</em>

**Comment by surjithctly** [comment]
How about adding `border-3` when `border-2` is not enough and `border-4` is too much.

**Reply by RobinMalfait** [reply]
Hey! Thank you for your suggestion!
Much appreciated! 🙏

You can extend your config if you want this behaviour: https://tailwindcss.com/docs/border-width#border-widths

**Comment by TaylorWatson** [comment]
Is there docs for v3? Even if uncompleted and possibly old information, so people who want to just mess around can at least somewhat navigate in the dark on new features?

**Comment by yevheniimoskalenko1** [comment]
![Снимок экрана 2021-12-02 в 16 28 46](https://user-images.githubusercontent.com/94050616/144441470-07f29428-72cb-4f53-adc8-59a78dbfb431.png)
@adamwathan  I have error with typography

**Comment by kkgthb** [comment]
Could there be more than `text-decoration-color` for `text-decoration`?  Like sizing & placement?  It'd be nice to be able to do tricks along the lines of [this](https://katiekodes.com/link-underline-grow-tailwind-css/) in native Tailwind.  I thought maybe I saw some energy in that direction:  https://github.com/tailwindlabs/tailwindcss/discussions/2050#discussioncomment-665373

**Reply by reinink** [reply]
Yes 😎 

[Text Decoration Color](https://tailwindcss.com/docs/text-decoration-color)
[Text Decoration Style](https://tailwindcss.com/docs/text-decoration-style)
[Text Decoration Thickness](https://tailwindcss.com/docs/text-decoration-thickness)
[Text Underline Offset](https://tailwindcss.com/docs/text-underline-offset)

**Comment by spencerwmiles** [comment]
Was the release of V3 intentional or some sort of error? By the sounds of this announcement it was not intended to drop for a couple more weeks but it's now available via @latest tag as of ~2 hours ago.

Just wanting to be sure before locking in V3 related changes in a project right now.

**Reply by dcastil** [reply]
I think this was intentional. See this Tweet from Adam a couple days ago. https://twitter.com/adamwathan/status/1467915766343348228
